usersession/agent/rest_api.go

```diff
@@ -349,3 +355,50 @@ func postRefreshFinishedNotification(c *Command, r *http.Request) Response {
 	}
 	return SyncResponse(nil)
 }
+
+func postRebootRequiredNotification(c *Command, r *http.Request) Response {
+	if ok, resp := validateJSONRequest(r); !ok {
+		return resp
+	}
+
+	decoder := json.NewDecoder(r.Body)
+
+	var rebootRequired client.RebootRequiredInfo
+	if err := decoder.Decode(&rebootRequired); err != nil {
+		return BadRequest("cannot decode request body into reboot required notification info: %v", err)
+	}
+
+	// Note that since the connection is shared, we are not closing it.
+	if c.s.bus == nil {
+		return SyncResponse(&resp{
+			Type:   ResponseTypeError,
+			Status: 500,
+			Result: &errorResult{
+				Message: "cannot connect to the session bus",
+			},
+		})
+	}
+
+	summary := fmt.Sprintf(i18n.G("Update available for %q."), rebootRequired.InstanceName)
+	body := i18n.G("Restart the system to update now.")
+	hints := []notification.Hint{
+		notification.WithDesktopEntry("io.snapcraft.SessionAgent"),
+		notification.WithUrgency(notification.LowUrgency),
+	}
+
+	msg := &notification.Message{
+		Title: summary,
+		Body:  body,
+		Hints: hints,
+	}
+	if err := c.s.notificationMgr.SendNotification(notification.ID(rebootRequired.InstanceName), msg); err != nil {
```

Comment:
Do we want one reboot notification, or one per snap that can trigger a reboot notification? If we only want one, then a fixed notification ID would make sense.

Also, is this notification meant to be distinct from the other notifications generated by the session agent? If so, we should be prefixing the IDs.

Reply:
In the current implementation it's one notification per snap. In practice I don't expect you'd have multiple snap updates though.

Added the prefix.